wordpress-mobile/WordPress-iOS
# Site Domains: Modernize UI 2/2 - Integrate status and navigation to domain details

Fixes #https://github.com/wordpress-mobile/WordPress-iOS/issues/22262

Continuation of https://github.com/wordpress-mobile/WordPress-iOS/pull/22294

## Description

Use `all-domains` endpoint instead of `sites/%d/domains` to load information needed to display status and open domain details.

## Solution

1. Use `DomainsService` to fetch domains and filter to show domains for a specific site
2. Cache fetched domains to preload them when opening All Domains
3. Add loading and error views
4. Add NavigationLink to open domain details web view

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/8c838b67-7e70-4049-b0e9-19343aadc0b5

## To test:

1. Check `Menu -> Domains` UI:
- With an account that has a domain (domains)
- With an account that only has a free domain
2. Confirm Status indicator appears
3. Confirm Status indicator matches the one on All Domains screen
4. Confirm tapping on paid domains opens Domain Management

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


UI Changes testing checklist:
